### DISCUSSION TITLE
Wrong content type on Windows when using StaticFiles

### DISCUSSION THREAD

**Discussion by ZeroIntensity** [encode/starlette#1568] [Potential Issue] [answered]
i'm using fastapi, but this issue originates from starlette.

when using `StaticFiles` on windows, the `Content-Type` header for js files isn't `application/javascript`, but instead `text/plain`. code below to reproduce:

```py
import uvicorn
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles  # same as starlette.StaticFiles

app = FastAPI()

app.mount("/static", StaticFiles(directory="./"), name="static")  # put something like a blank "a.js" in the current directory to test

if __name__ == '__main__':
    uvicorn.run(app, port = 8080)
```
```py
import requests

print(requests.get('http://127.0.0.1:8080/static/a.js').headers['content-type'])
```
i tested the above on linux (ubuntu 20.04 lts), and it works just fine. this seems to only occur on windows

this also seems to be only for js files, i tested it with html and css and it returned the correct content type.

**Comment by Kludex** [comment, marked as answer]
This was already mentioned on https://github.com/encode/starlette/issues/829.
